### DISCUSSION TITLE
Background images

### DISCUSSION THREAD

**Discussion by Intiiii** [TomSchimansky/CustomTkinter#2331] [Q&A]
Hi, I want to make a app that has a background image and I figured something out alr but that one doesn't use grids, is it possible to do it with grids?

this is my code:

`from customtkinter import *
import actions
from PIL import Image
import pywinstyles
import tkinter

set_appearance_mode('system')
set_default_color_theme('blue')
default_border_color = ''
rel_path = os.path.join(os.path.dirname(__file__))

class App(CTk):
    def __init__(self):
        super().__init__()
        
        self.title('Coornhert Connect')
        self.geometry('750x490')
        
        frame = CTkFrame(
            self,
            width=750,
            height=490,
            bg_color = 'transparent'
            )
        frame.pack()

        image_label = CTkLabel(frame, text="", image=CTkImage(Image.open("image_4.png"), size=(750,490)))
        image_label.pack(expand=True, fill="both")

        #portal
        button1 = CTkButton(
            frame, 
            text = 'hi',
            width=40.0,
            height=40.0,
            command =  actions.homebutton,
            bg_color="#000001",
            border_color = default_border_color,
            border_width = 1
        )
        button1.place(
            x=10.0,
            y=10.0
        )
        pywinstyles.set_opacity(button1, color="#000001")

        #outlook
        button2 = CTkButton(
            frame, 
            width=40.0,
            height=40.0,
            text = 'mail',    
            command = actions.mail,
            bg_color="#000001",
            border_color = default_border_color,
            border_width = 1
        )
        button2.place(
            x=10.0,
            y=60.0
        )
        pywinstyles.set_opacity(button2, color="#000001")

        #somtoday
        button3 = CTkButton(
            frame, 
            width=40.0,
            height=40.0,
            text = 'som',    
            command = actions.somtoday,
            bg_color="#000001",
            border_color = default_border_color,
            border_width = 1
        )
        button3.place(
            x=10.0,
            y=110.0
        )
        pywinstyles.set_opacity(button3, color="#000001")
        
        #zermelo
        button4 = CTkButton(
            frame, 
            width=40.0,
            height=40.0,
            text = 'zer',    
            command = actions.zermelo,
            bg_color="#000001",
            border_color = default_border_color,
            border_width = 1
        )
        button4.place(
            x=10.0,
            y=160.0
        )
        pywinstyles.set_opacity(button4, color="#000001")
        
        #liber
        button5 = CTkButton(
            frame, 
            width=40.0,
            height=40.0,
            text = 'Lib',    
            command = actions.liber,
            bg_color="#000001",
            border_color = default_border_color,
            border_width = 1
        )
        button5.place(
            x=10.0,
            y=210.0
        )
        pywinstyles.set_opacity(button5, color="#000001")
        
        #schooldocs
        button6 = CTkButton(
            frame, 
            width=40.0,
            height=40.0,
            text = 'doc',    
            command = actions.schooldocs,
            bg_color="#000001",
            border_color = default_border_color,
            border_width = 1
        )
        button6.place(
            x=10.0,
            y=260.0
        )
        pywinstyles.set_opacity(button6, color="#000001")
        
        #DWO
        button7 = CTkButton(
            frame, 
            width=40.0,
            height=40.0,
            text = 'dwo',    
            command = actions.DWO,
            bg_color="#000001",
            border_color = default_border_color,
            border_width = 1
        )
        button7.place(
            x=10.0,
            y=310.0
        )
        pywinstyles.set_opacity(button7, color="#000001")
        
        #catwise
        button8 = CTkButton(
            frame, 
            width=40.0,
            height=40.0,
            text = 'Cat',    
            command = actions.Catwise,
            bg_color="#000001",
            border_color = default_border_color,
            border_width = 1
        )
        button8.place(
            x=10.0,
            y=360.0
        )
        pywinstyles.set_opacity(button8, color="#000001")
        
        #wis-actief
        button9 = CTkButton(
            frame, 
            width=40.0,
            height=40.0,
            text = 'W-A',    
            command = actions.WisAct,
            bg_color="#000001",
            border_color = default_border_color,
            border_width = 1
        )
        button9.place(
            x=10.0,
            y=410.0
        )
        pywinstyles.set_opacity(button9, color="#000001")
        
        #searchbar
        searchentry = CTkEntry(
            frame, 
            placeholder_text="search on google",
            width=223.0,
            height=40.0,
            bg_color="#000001",
            border_color = 'white',
            border_width = 0,
            corner_radius = 15
            )
        searchentry.place(
            x=244.0,
            y=205.0
        )
        pywinstyles.set_opacity(searchentry, color="#000001")
        
        def clear_entry():
            searchentry.delete(0, 'end')
        def on_enter_pressed(event):
            actions.search(searchentry.get())
            searchentry.delete(0, 'end')

        searchentry.bind("<Return>", on_enter_pressed)
            
        searchbutton = CTkButton(
            frame, 
            width=40.0,
            height=40.0,
            text = '.',
            command = lambda: [actions.search(searchentry.get()),clear_entry()],
            bg_color="#000001",
            fg_color = 'white',
            border_color = default_border_color,
            border_width = 1,
            corner_radius = 15
        )
        searchbutton.place(
            x=467.0,
            y=205.0
        )
        pywinstyles.set_opacity(searchbutton, color="#000001")
        
app = App()
app.mainloop()`

**Comment by dipeshSam** [comment]
Grid geometry manager is **not** designed to work as a background for the same window.

To implement background facility, user is recommended to use `place` or `pack` geometry managers.

User can use companion methods like `lower()`, `lift()` or `raise()` and attributes like `after` and/or `before`  to lift up or send back the childrens with respect to the parent.

Regards.
